documentation/configuration.markdown

```diff
@@ -144,8 +144,13 @@ The init configuration file location depends on your operating system.
     ```
     Jun 01 19:58:16 myhost puppetserver[24001]: Unable to find specified JRUBY_JAR:/opt/puppetlabs/server/apps/puppetserver/jruby-wrong.jar
     ```
+2.  Set other configuration options for better performance
 
-2.  Restart the `puppetserver` service.
+    To ensure good performance with JRuby 9k, the following settings should also be updated:
+    * In `puppetserver.conf`, `jruby-puppet.compile-mode` should be set to `jit`. This setting must be switched back when reverting to 1.7.
+    * Add `-XX:ReservedCodeCacheSize=512m` to `JAVA_ARGS`, typically defined in `/etc/sysconfig/puppetserver`. This scales up the JVM's code cache to the size needed by JRuby 9k. This setting also behaves well when using JRuby 1.7, so it is safe to leave this setting on if you are switching back and forth.
```

Comment:
What does this setting do in 1.7 if enabled?

If this setting is safe in 1.7, can Server set it by default so the user doesn't need to?

Reply:
We haven't updating our packaging to do so yet, but we have encouraged people to start turning 9k on. I think we are planning to set this by default going forward. When we do, we can update the docs for that release.